invite/test/invite_bot.test.ts

```diff
@@ -0,0 +1,302 @@
+import {GitHub} from '../src/github';
+
+describe('Invite Bot', () => {
+  let inviteBot: InviteBot;
+
+  beforeEach(() => {
+    inviteBot = new InviteBot(/*client=*/ null, 'test_repo');
+
+    jest.spyOn(GitHub.prototype, 'inviteUser');
+    jest.spyOn(GitHub.prototype, 'addComment');
+    jest.spyOn(InvitationRecord.prototype, 'recordInvite');
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('parseMacros', () => {
+    describe('comments without macros', () => {
+      it('returns an empty list', () => {
+        const comment = 'random comment';
+        expect(inviteBot.parseMacros(comment)).toEqual({});
+      });
+
+      it('ignores macros without usernames', () => {
+        const comment = '/invite not_a_tag';
+        expect(inviteBot.parseMacros(comment)).toEqual({});
+      });
+    });
+
+    describe('comments with `/invite @user`', () => {
+      it('returns list including Invite action for @user', () => {
+        const comment = '/invite @someone';
+        expect(inviteBot.parseMacros(comment)).toEqual({
+          someone: InviteAction.INVITE,
+        });
+      });
+
+      it('handles multiple `/invite` macros', () => {
+        const comment = '/invite @someone and /invite @someone_else';
+        expect(inviteBot.parseMacros(comment)).toEqual({
+          someone: InviteAction.INVITE,
+          someone_else: InviteAction.INVITE,
+        });
+      });
+    });
+
+    describe('comments with `/tryassign @user`', () => {
+      it('returns list including InviteAndAssign action for @user', () => {
+        const comment = '/tryassign @someone';
+        expect(inviteBot.parseMacros(comment)).toEqual([
+          {
+            someone: InviteAction.INVITE_AND_ASSIGN,
+          },
+        ]);
+      });
+
+      it('handles multiple `/tryassign` macros', () => {
+        const comment = '/tryassign @someone and /tryAssign @someone_else';
+        expect(inviteBot.parseMacros(comment)).toEqual([
+          {
+            someone: InviteAction.INVITE_AND_ASSIGN,
+            someone_else: InviteAction.INVITE_AND_ASSIGN,
+          },
+        ]);
+      });
+    });
+
+    it('handles mix of `/invite` and `/tryassign` macros', () => {
+      const comment = '/invite @someone and /tryAssign @someone_else';
+      expect(inviteBot.parseMacros(comment)).toEqual([
+        {
+          someone: InviteAction.INVITE,
+          someone_else: InviteAction.INVITE_AND_ASSIGN,
+        },
+      ]);
+    });
+  });
+
+  describe('tryInvite', () => {
+    const newInvite: Invite = {
+      username: 'someone',
+      repo: 'test_repo',
+      issue_number: 1337,
+      action: InviteAction.INVITE,
+    };
+
+    describe('when the user has a pending invite from the bot', () => {
+      beforeEach(() => {
+        inviteBot.record.recordInvite({
+          username: 'someone',
+          repo: 'test_repo',
+          issue_number: 42,
+          action: InviteAction.INVITE,
```

Comment:
You know how you see a word dozens of times and you suddenly begin to doubt whether it's spelled right? Cause that just happened to me with INVITE

Reply:
LOL yep